src/js/tech/html5.js

```diff
@@ -783,6 +783,62 @@ class Html5 extends Tech {
     this.setSrc(src);
   }
 
+  /**
+   * Add a <source> element to the <video> element.
+   *
+   * @param {string} srcUrl
+   *        The URL of the video source.
+   *
+   * @param {string} [mimeType]
+   *        The MIME type of the video source. Optional but recommended.
+   */
+  addSourceElement(srcUrl, mimeType) {
+    if (!srcUrl) {
+      log.error('Invalid source URL.');
+      return;
+    }
+
+    const sourceAttributes = { src: srcUrl };
+
+    if (mimeType) {
+      sourceAttributes.type = mimeType;
+    }
+
+    const sourceElement = Dom.createEl('source', {}, sourceAttributes);
+
+    this.el_.appendChild(sourceElement);
+  }
+
+  /**
+   * Remove a <source> element from the <video> element by its URL.
+   *
+   * @param {string} srcUrl
+   *        The URL of the source to remove.
+   *
+   * @return {boolean}
+   *         Returns true if the source element was successfully removed, false otherwise.
+   */
+  removeSourceElement(srcUrl) {
+    if (!srcUrl) {
+      log.error('Source URL is required to remove the source element.');
+      return false;
```

Comment:
Is there a reason this function returns a boolean but `addSourceElement` does not?

Reply:
Good call, both probably should. The reason I didn't have `addSourceElement()` return a boolean is because originally I hadn't added that early return so a source element would always be added.

Reply:
Updated